### PR TITLE
misc(build): generate report variants in deployment

### PR DESCRIPTION
**Pull request by paulirish**
Currently our builder just gives us a single sample LHR report. But we'd like to see it in other languages and in a more devtools-like view.

e.g. https://lighthouse-git-deployvariants.googlechrome.now.sh/

image

I have another PR waiting that adds an LHR/report with lots of errors.

### CODE REVIEW

File: lighthouse-core/runner.js [hunk 1]
```diff
@@ -237,6 +237,7 @@ class Runner {
         gatherMode: undefined,
         auditMode: undefined,
         output: undefined,
+        channel: undefined,
```
Comment (paulirish) [inline lighthouse-core/runner.js]: mostly unrelated.  just a driveby.

File: lighthouse-core/scripts/build-report-for-autodeployment.js [hunk 2]
```diff
@@ -22,8 +26,26 @@ lhr.categories['lighthouse-plugin-someplugin'] = {
...
+    'espanol': swapLocale(lhr, 'es').lhr,
+    'arabic': swapLocale(lhr, 'ar').lhr,
+  };
+
+  mkdirp(DIST);
+
+  // Generate and write reports
+  Object.entries(filenameToLhr).forEach(([filename, lhr]) => {
+    let html = ReportGenerator.generateReportHtml(lhr);
+    for (const variant of ['', '-devtools']) {
```
Comment (paulirish) [inline lighthouse-core/scripts/build-report-for-autodeployment.js]: in the future, PSI will be another report variant.

File: package.json [hunk 3]
```diff
@@ -152,6 +152,7 @@
     "jsonlint-mod": "^1.7.4",
     "lighthouse-logger": "^1.2.0",
     "lodash.isequal": "^4.5.0",
+    "lodash.set": "^4.3.2",
```
Comment (patrickhulce) [inline package.json]: bundlephobia says this is 2KB minified+gzipped, so basically a nonfactor for folks already invoking `npm install lighthouse` 😆

File: package.json [hunk 4]
```diff
@@ -47,6 +47,7 @@
     "chrome": "node lighthouse-core/scripts/manual-chrome-launcher.js",
     "fast": "yarn start --emulated-form-factor=none --throttlingMethod=provided",
     "deploy-viewer": "yarn build-viewer --deploy",
+    "now-build": "node lighthouse-core/scripts/build-report-for-autodeployment.js",
```
Comment (patrickhulce) [inline package.json]: neat!

File: lighthouse-core/scripts/build-report-for-autodeployment.js [hunk 5]
```diff
@@ -22,8 +26,26 @@ lhr.categories['lighthouse-plugin-someplugin'] = {
...
-console.log('🕒 Generating report for sample_v2.json...');
-const html = ReportGenerator.generateReport(lhr, 'html');
-const filename = path.join(__dirname, '../../dist/index.html');
-fs.writeFileSync(filename, html, {encoding: 'utf-8'});
-console.log('✅', filename, 'written.');
+(async function() {
+  const filenameToLhr = {
+    'english': lhr,
+    'espanol': swapLocale(lhr, 'es').lhr,
+    'arabic': swapLocale(lhr, 'ar').lhr,
```
Comment (patrickhulce) [inline lighthouse-core/scripts/build-report-for-autodeployment.js]: can we do `en-xa-local` once #9192 lands? 😃

Comment (paulirish) [reply lighthouse-core/scripts/build-report-for-autodeployment.js]: yup. definitely will.
